<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
  <head>
    <title>A beginners guide to Call SNPs and indels: Part II</title>
    <meta charset="utf-8" />
    <meta name="author" content="俞丽佳" />
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/default-fonts.css" rel="stylesheet" />
    <script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
    <script src="libs/viz-0.3/viz.js"></script>
    <link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
    <script src="libs/grViz-binding-1.0.1/grViz.js"></script>
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# A beginners guide to Call SNPs and indels: Part II
## <em>Mark Duplicates and Base (Quality Score) Recalibration</em>
### 俞丽佳
### 2019/07/31<br/>(updated: 2019-08-04)

---



# Outline

1. Review
2. Mark Duplicates
3. Base (Quality Score) Recalibration

---


### Q1: How to create a new directory with multiple subdirectories? 

```bash
cd ~
mkdir ./project
mkdir ./project/out
mkdir ./tmp
mkdir ./project/src
mkdir ./project/in
```

```bash
mkdir -p ./project/{in,out,src,tmp}
```

---

### Q2: What is src stand for ?

```
.
├── in
├── out
├── src
└── tmp

```

- in: input (.fq.gz)
- out: output (.bam)
- src: source code (script.sh)


Always run the code from `src` directory.


---

### Q3. How to use pipe?

```bash
java -Xmx4G -jar trimmomatic-0.36.jar PE \
-threads 4 \
-phred64 \
../in/normal_R1.fq.gz \
../in/normal_R2.fq.gz \
../out/normal_R1.trimed.fq.gz \
../out/normal_R1.unpaired.fq.gz \
../out/normal_R2.trimed.fq.gz \
../out/normal_R2.unpaired.fq.gz TOPHRED33 |\
bwa mem -M -t 4\
/home/admin/database/reference/hg19/ucsc.hg19.fasta \
../out/normal_R1.trimed.fq.gz \ 
../out/normal_R2.trimed.fq.gz |\
samtools sort -o ../out/normal.sorted.bam -

```

---

### Q3. How to use pipe?

```bash
$BWA mem ... | $SAMTOOLS sort ...
command_1 | command_2 | command_3 | .... | command_N
```

Pipe is used to combine two or more commands, and in this, the output of one command acts as input to another command, and this command’s output may act as input to the next command and so on.

A pipe is a form of redirection (transfer of &lt;u&gt;standard output&lt;/u&gt; to some other destination) that is used in Linux and other Unix-like operating systems to send the output of one command/program/process to another command/program/process for further processing. 


.footnote[
[1][Piping in Unix or Linux](https://www.geeksforgeeks.org/piping-in-unix-or-linux/)
]
---

# Pipeline 

<div id="htmlwidget-526012b67b28b13a74b3" style="width:504px;height:504px;" class="grViz html-widget"></div>
<script type="application/json" data-for="htmlwidget-526012b67b28b13a74b3">{"x":{"diagram":"digraph flowchart {\n      # node definitions with substituted label text\n      node [fontname = Helvetica, shape = rectangle]        \n      tab1 [label = \"Quality Control: fastQC\"]\n      tab2 [label = \"Phred 64 to 33: Trimmomatic\"]\n      tab3 [label = \"Map to Reference: Burrows-Wheeler Aligner\"]\n      tab4 [label = \"Mark Duplicates: ??\"]\n      tab5 [label = \"Base Recalibration: ??\"]\n\n      # edge definitions with the node IDs\n      tab1 -> tab2 -> tab3 -> tab4 -> tab5;\n      }","config":{"engine":"dot","options":null}},"evals":[],"jsHooks":[]}</script>

---

# GATK workflow

![](./2019-07-31-A_beginners_guide_to_Call_SNPs_and_indels_Part_II_files/figure/GATKworkflows.png)

.footnote[
[1][Introduction to Variant Discovery with GATK Best Practices](https://qcb.ucla.edu/wp-content/uploads/sites/14/2016/03/GATKwr12-1-GATK_primer.pdf)
]

---

# SAM/BAM format


.center[
![](https://us.v-cdn.net/5019796/uploads/editor/f4/uuzmf2cbau1y.png)
]

SAM stands for &lt;u&gt;Sequence Alignment Map&lt;/u&gt; and is described in the standard specification [here](http://samtools.github.io/hts-specs/). BAM and CRAM are both compressed forms of SAM; BAM (for Binary Alignment Map) is a lossless compression while CRAM can range from lossless to lossy depending on how much compression you want to achieve (up to very much indeed). 

.footnote[
[1] [SAM / BAM / CRAM - Mapped sequence data formats](https://gatkforums.broadinstitute.org/gatk/discussion/11014/sam-bam-cram-mapped-sequence-data-formats)

[2] [SAM/BAM and related specifications](http://samtools.github.io/hts-specs/)
]

---

# SAM/BAM format: header


```bash
samtools view -H read_data.bam
```


&gt; @HD VN:1.5  GO:none SO:coordinate

&gt; @SQ SN:20   LN:63025520 

&gt; @RG ID:H0164.2  PL:illumina PU:H0164ALXX140820.2 LB:Solexa-272222
&gt; PI:0 DT:2014-08-20T00:00:00-0400 SM:NA12878 CN:BI

&gt; @PG [program group metadata]



The header should contain read groups with **sample names** (SM:NA12878). 

The presence of the @RG tags indicate the presence of read groups. Each read group has a SM tag, indicating the sample from which the reads belonging to that read group originate.


.footnote[
[1] [SAM / BAM / CRAM - Mapped sequence data formats](https://gatkforums.broadinstitute.org/gatk/discussion/11014/sam-bam-cram-mapped-sequence-data-formats)
]

---

# SAM/BAM format: CIGAR

.center[
![](./2019-07-31-A_beginners_guide_to_Call_SNPs_and_indels_Part_II_files/figure/CIGAR.png)]



[1][Introduction to Variant Discovery with GATK Best Practices](https://qcb.ucla.edu/wp-content/uploads/sites/14/2016/03/GATKwr12-1-GATK_primer.pdf)
---

# Genome Analysis Toolkit

.center[
![](./2019-07-31-A_beginners_guide_to_Call_SNPs_and_indels_Part_II_files/figure/gatk.png)]

- the toolkit offers a wide variety of tools with a primary focus on variant discovery and genotyping. Its powerful processing engine and high-performance computing features make it capable of taking on projects of any size.

- Mutect2 / MuTect2 / MuTect
- Picard ( a set of command line tools for manipulating high-throughput sequencing (HTS) data and formats such as SAM/BAM/CRAM and VCF. )
- GenomeAnalysisTK (&lt;4.0) / gatk (4.0)


---

# Add header

```bash
/home/admin/software/gatk-4.1.0.0/gatk AddOrReplaceReadGroups \
--VALIDATION_STRINGENCY=SILENT \
--RGLB=hg19 \
--RGPL=illumina \
--RGPU=hg19PU \
--RGSM=cancer \
-I=../out/cancer.sorted.bam \
-O=../out/cancer.addheader.bam 
```

- RGSM is Read-Group sample name (cancer/normal)


.footnote[
[1] [AddOrReplaceReadGroups (Picard)](https://software.broadinstitute.org/gatk/documentation/tooldocs/4.0.1.1/picard_sam_AddOrReplaceReadGroups.php)
]

---


# Mark duplicates to mitigate	duplication	artifacts	
 

.center[
![](./2019-07-31-A_beginners_guide_to_Call_SNPs_and_indels_Part_II_files/figure/duplication.png) 
]



---

# MarkDuplicates on Spark (GATK4)

```bash
/home/admin/software/gatk-4.1.0.0/gatk MarkDuplicatesSpark \
-I input.bam \
-O marked_duplicates.bam \
-M marked_dup_metrics.txt \
--conf 'spark.executor.cores=1'
```

This is a Spark implementation of Picard MarkDuplicates that allows the tool to be run in parallel on multiple cores on a local machine

.footnote[
[1] [MarkDuplicatesSpark](https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_hellbender_tools_spark_transforms_markduplicates_MarkDuplicatesSpark.php)
]
---

# Local realignment around indels corrects mapping errors

.center[&lt;img src="2019-07-31-A_beginners_guide_to_Call_SNPs_and_indels_Part_II_files/figure/indel_1.png"&gt;]


| position | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 
|------------|---|---|---|---|---|---|---|---|---|---|---|---|---|---|
| Reference| C | A | T | T | T | T | T | T | T | C | T | A | A | G |
| group 1 |...| C | A | T | T | T | T | T | T |C | T | A | A | G |
| group 2 | C | A | T | T | T | T | T | T |C | T | A | A | G |...|

---

# Local realignment around indels corrects mapping errors

.center[&lt;img src="2019-07-31-A_beginners_guide_to_Call_SNPs_and_indels_Part_II_files/figure/indel_2.png" width="550"&gt;]

| position | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 
|------------|---|---|---|---|---|---|---|---|---|---|---|---|---|---|
| Reference| C | A | T | T | T | T | T | T | T | C | T | A | A | G |
| group 1 | C | A | - | T | T | T | T | T | T |C | T | A | A | G |
| group 2 | C | A | - | T | T | T | T | T | T |C | T | A | A | G |
---

# RealignerTargetCreator and IndelRealigner (GATK3, retired)

```bash
 java -jar GenomeAnalysisTK.jar \
   -T RealignerTargetCreator \
   -R reference.fasta \
   -I input.bam \
   --known indels.vcf \
   -o forIndelRealigner.intervals

 java -jar GenomeAnalysisTK.jar \
   -T IndelRealigner \
   -R reference.fasta \
   -I input.bam \
   -known indels.vcf \
   -targetIntervals intervalListFromRTC.intervals \
   -o realignedBam.bam

```


.footnote[
[1] [RealignerTargetCreator](https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_indels_RealignerTargetCreator.php)

[2] [IndelRealigner](https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_indels_IndelRealigner.php)
]

---

# Base (Quality Score) Recalibration
 
 
![](./2019-07-31-A_beginners_guide_to_Call_SNPs_and_indels_Part_II_files/figure/Recalibration.png) 

[1][Introduction to Variant Discovery with GATK Best Practices](https://qcb.ucla.edu/wp-content/uploads/sites/14/2016/03/GATKwr12-1-GATK_primer.pdf)
 
---

# Base (Quality Score) Recalibration: BaseRecalibrator

```bash
/home/admin/software/gatk-4.1.0.0/gatk BaseRecalibrator \
-I my_reads.bam \
-R reference.fasta \
--known-sites dbsnp_138.hg19.vcf \
--known-sites Mills_and_1000G_gold_standard.indels.hg19.sites.vcf \
--known-sites 1000G_phase1.indels.hg19.sites.vcf \
--known-sites 1000G_phase1.snps.high_confidence.hg19.sites.vcf \
-L ../in/Illumina.bed \
-O recal_data.table
```

ExAc, gnomAD, or dbSNP resources can be used as known sites of variation.

We assume that all reference mismatches we see are therefore errors and indicative of poor base quality. 

.footnote[
[1] [BaseRecalibrator](https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_hellbender_tools_walkers_bqsr_BaseRecalibrator.php)
]

---

# Resource bundle

- `dbSNP 138`
   + evaluation of dbSNP rate and Ti/Tv values at novel sites.
   


- `Mills_and_1000G_gold_standard`
   + best set of known indels to be used for local realignment


- `1000G_phase1.indels`
   + best set of known indels to be used for local realignment
   


- `1000G_phase1.snps.high_confidence`
   + genotype refinement


The GATK resource bundle is a collection of standard files for working with human resequencing data with the GATK.

.footnote[
[1] [Resource Bundle: Reference materials for human analysis](https://software.broadinstitute.org/gatk/documentation/article.php?id=11017)
]


---

# Base (Quality Score) Recalibration: ApplyBQSR


```bash

/home/admin/software/gatk-4.1.0.0/gatk ApplyBQSR \
-R reference.fasta \
-I input.bam \
--bqsr-recal-file recal_data.table \
-L ../in/Illumina.bed \
-O output.bam
   
```

It recalibrates the base qualities of the input reads based on the recalibration table produced by the BaseRecalibrator tool, and outputs a recalibrated BAM or CRAM file.


.footnote[
[1] [ApplyBQSR](https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_hellbender_tools_walkers_bqsr_ApplyBQSR.php)
]

---

# Quitz

1.Could you write the commands of  Mark Duplicates and Base (Quality Score) Recalibration by yourself?

Please write your codes in a script file (script.sh).

---

# Resource

This section is based on the GATK 4 Best Practices workflow at [https://software.broadinstitute.org/gatk/best-practices/](https://software.broadinstitute.org/gatk/best-practices/)
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();</script>

<script>
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
